### ISSUE TITLE
rand(::Normal, ::Dims) doesn't work

### ISSUE THREAD

**Issue by johnnychen94** [JuliaStats/Distributions.jl#795]
I found following generic codes https://github.com/JuliaStats/Distributions.jl/blob/117894c129420b7243a1e2d10ad82dc4bb376df1/src/genericrand.jl#L43-L47

but `rand(::Normal, ::Dims)` doesn't work though...

```julia
julia> img = ones(5,5);

julia> rand(Normal(),size(img))
ERROR: MethodError: no method matching Array{Float64,1}(::UndefInitializer, ::Tuple{Int64,Int64})
Closest candidates are:
  Array{Float64,1}(::UndefInitializer, ::Tuple{Vararg{Int64,N}}) where {T, N} at boot.jl:406
  Array{Float64,1}(::UndefInitializer, ::Tuple{Vararg{Integer,N}}) where {T, N} at sysimg.jl:155
  Array{Float64,1}(::UndefInitializer, ::Int64) where T at boot.jl:394
  ...
Stacktrace:
 [1] rand(::Normal{Float64}, ::Tuple{Int64,Int64}) at /Users/jc/.julia/packages/Distributions/WHjOk/src/univariates.jl:169
 [2] top-level scope at none:0

julia> rand(Normal(),size(img)...)
5×5 Array{Float64,2}:
  0.462929  -0.063203   2.63356   -0.996424  -0.366942
 -1.01405    0.218144   0.180686  -0.720094  -1.04636
 -0.442738  -1.26272   -0.248081   0.89015   -0.273845
 -0.127492  -0.851691  -1.99012    0.688425  -0.512067
 -2.64416   -1.18138    0.483727  -0.780079   0.0627713
```

Is this a deliberately designed behavior?

**Comment by richardreeve** [comment]
@matbesancon This is now fixed by #830 and can be closed:
```julia
julia> img = ones(5,5);

julia> rand(Normal(),size(img))
5×5 Array{Float64,2}:
 -0.839027   0.431422   0.408396   -0.757633  -0.0633746
  0.311111   0.583708  -0.0504512  -1.72976    1.33694
  2.29509    0.963272  -0.693654    0.795949  -0.0731486
 -2.26709    0.458791  -1.77383     0.670062  -0.745464
  0.529966  -0.522337   0.120766    0.550852  -1.22006

julia> rand(Normal(),size(img)...)
5×5 Array{Float64,2}:
 -0.0531773   0.567695  -1.80303     1.73517    1.12941
 -0.165136    0.763499  -1.28905     1.29992   -1.34853
 -2.11537     0.378599  -0.33527     0.206364   1.24029
 -0.066768   -0.645597   0.0704676  -1.00886   -0.0556471
  1.22246    -0.664646   0.341794   -0.850056   0.75939
```

**Comment by matbesancon** [comment]
:tada:
